x/upgrade/keeper/keeper.go

```diff
@@ -100,19 +107,45 @@ func (k Keeper) SetUpgradedClient(ctx sdk.Context, cs ibcexported.ClientState) e
 	}
 
 	store := ctx.KVStore(k.storeKey)
-	store.Set(types.UpgradedClientKey(), bz)
+	store.Set(types.UpgradedClientKey(upgradeHeight), bz)
 	return nil
 }
 
 // GetUpgradedClient gets the expected upgraded client for the next version of this chain
-func (k Keeper) GetUpgradedClient(ctx sdk.Context) (ibcexported.ClientState, error) {
+// along with the planned upgrade height
+func (k Keeper) GetUpgradedClient(ctx sdk.Context) (ibcexported.ClientState, int64, error) {
 	store := ctx.KVStore(k.storeKey)
-	bz := store.Get(types.UpgradedClientKey())
-	if len(bz) == 0 {
-		return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "no upgraded client in store")
+	iterator := sdk.KVStorePrefixIterator(store, []byte(types.KeyUpgradedClient))
+	var (
+		count, height int
+		bz            []byte
+	)
+
+	defer iterator.Close()
+	for ; iterator.Valid(); iterator.Next() {
+		count++
+
+		keySplit := strings.Split(string(iterator.Key()), "/")
+		var err error
+		height, err = strconv.Atoi(keySplit[len(keySplit)-1])
+		if err != nil {
+			return nil, 0, sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "could not parse upgrade height from key: %s", err)
```

Comment:
why `sdk.ErrInvalidRequest`? It'd make more sense to me to Wrap(err, "could not parse upgrade height")

Reply:
Following the format of the rest of this file